modules/heat_conduction/src/interfacekernels/ThinLayerHeatTransfer.C

```diff
@@ -0,0 +1,115 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ThinLayerHeatTransfer.h"
+
+registerMooseObject("HeatConductionApp", ThinLayerHeatTransfer);
+
+InputParameters
+ThinLayerHeatTransfer::validParams()
+{
+  InputParameters params = InterfaceKernel::validParams();
+  params.addClassDescription("Model heat transfer across a thin domain with an interface.");
+  params.addParam<MaterialPropertyName>(
+      "specific_heat",
+      "specific_heat",
+      "Property name of the specific heat material property of the thin layer");
+
+  params.addParam<MaterialPropertyName>(
+      "density", "density", "Property name of the density material property of the thin layer");
+
+  params.addParam<MaterialPropertyName>(
+      "heat_source",
+      "heat_source",
+      "Property name of the heat source material property of the thin layer");
```

Comment:
At some point it would be nice to have these strings as constants in `HeatConductionNames` but that doesn't have to be today